package/MDAnalysis/transformations/nojump.py

```diff
@@ -0,0 +1,158 @@
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""\
+No Jump Trajectory Unwrapping --- :mod:`MDAnalysis.transformations.nojump`
+=======================================================================================
+
+Unwraps the trajectory such that atoms never move more than half a periodic box length.
+The consequence of this is that particles will diffuse across periodic boundaries when
+needed. This unwrapping method is suitable as a preprocessing step to calculate
+molecular diffusion.
+The algorithm used is based on :cite:p:`Kulke2022`.
+
+.. autoclass:: NoJump
+
+"""
+import numpy as np
+import warnings
+
+from .base import TransformationBase
+from ..due import due, Doi
+from ..exceptions import NoDataError
+
+
+class NoJump(TransformationBase):
+    """
+    Returns transformed coordinates for the given timestep so that an atom
+    does not move more than half the periodic box size, and will move
+    across periodic boundary edges. The algorithm used is based on :cite:p:`Kulke2022`,
+    equation B6 for non-orthogonal systems.
+
+    Example
+    -------
+
+    To calculate diffusion, one of the first steps is to compute a trajectory
+    where the atoms do not cross the periodic boundary.
+
+    .. code-block:: python
+
+        transformation = NoJump()
+        u.trajectory.add_transformations(transformation)
+        for ts in u.trajectory:
+            print(ts.positions)
+
+    In this case, ``ts.positions`` will return the NoJump unwrapped trajectory.
+    To reverse the process, wrap the coordinates again.
+
+    Returns
+    -------
+    MDAnalysis.coordinates.timestep.Timestep
+
+    References
+    ----------
+    .. bibliography::
+        :filter: False
+        :style: MDA
+
+        Kulke2022
+
+    """
+
+    @due.dcite(
+        Doi("10.1021/acs.jctc.2c00327"),
+        description="Works through the orthogonal case for unwrapping, "
+        "and proposes the non-orthogonal approach.",
+        path=__name__,
+    )
+    def __init__(
+        self,
+        check_continuity=True,
+        max_threads=None,
+        # NoJump transforms are inherently unparallelizable, since
+        # it depends on the previous frame's unwrapped coordinates
+        parallelizable=False,
+    ):
+        super().__init__(max_threads=max_threads, parallelizable=parallelizable)
+        self.prev = None
+        self.old_frame = 0
+        self.older_frame = "A"
+        self.check_c = check_continuity
+
+    def _transform(self, ts):
+        if self.prev is None:
+            try:
```

Comment:
I think this should be seperated into two checks, one that checks `ts.data != None` and another that checks for invertibility as this will be easier for users.

Reply:
I've worked out what the transform should be for the [1::3] and [2::3] cases, and added more text to try and explain this. The "correct" answer is that the sqrt(3)/2 comes from the 60 degree angle set in the dimension. This makes a hexagonal unit cell, which has as a consequence that the origin for adjacent PBC images shifts by an unusual amount. Since in this particular test, I'm transforming a line *in unit cell coordinates*, the actual unwrapped trajectory is zig-zagging a bit along the hexagonal lattice.

@hmacdope , do we actually need to check that `ts.data != None`? The check you highlighted is actually testing if this is the first frame to be unwrapped or not. You can't unwrap the first frame. I have however noticed that the L and Linverse computations can be done together, before this test.